### PR TITLE
GO-1775 Fix removal of required blocks when pasting text

### PR DESCRIPTION
**Pull request by fat-fellow**
---

- [ ] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: core/block/editor/clipboard/paste.go [hunk 1]
```diff
@@ -292,7 +296,12 @@ func (p *pasteCtrl) insertUnderSelection() (err error) {
 
 func (p *pasteCtrl) removeSelection() {
 	for _, toRemove := range p.selIds {
-		p.s.Unlink(toRemove)
+		if toRemove != template.TitleBlockId &&
```
Comment (AnastasiaShemyakinskaya) [inline core/block/editor/clipboard/paste.go:299]: May be extract this condition to seperated function to make it clearer?

Comment (fat-fellow) [reply core/block/editor/clipboard/paste.go:299]: agreed

File: core/block/editor/clipboard/paste.go [hunk 2]
```diff
@@ -290,9 +291,18 @@ func (p *pasteCtrl) insertUnderSelection() (err error) {
 	})
 }
 
+func isObligatoryRelation(targetID string) bool {
```
Comment (AnastasiaShemyakinskaya) [inline core/block/editor/clipboard/paste.go:294]: Imho, `Required` is more common word than `Obligatory`

Comment (fat-fellow) [reply core/block/editor/clipboard/paste.go:294]: so be it

**Review by AnastasiaShemyakinskaya — APPROVED** [review]

**Pull request merged**